x-pack/plugins/ingest_manager/server/plugin.ts

```diff
@@ -90,12 +92,31 @@ const allSavedObjectTypes = [
   ENROLLMENT_API_KEYS_SAVED_OBJECT_TYPE,
 ];
 
+/**
+ * Callbacks supported by the Ingest plugin
+ */
+export type ExternalCallbacks = [
```

Comment:
Going forward if we need add additional callback, this type can be expanded. Example:

```typescript

export type ExternalCallbacks = 
  | [ 'datasourceCreate', (newDatasource: NewDatasource) => Promise<NewDatasource>]
  | [ 'datasourceUpdate', () => {}]
  | [ `agentConfigCreate', () => {}];
```